src/api/app/validators/workflow_event_filter_validator.rb

```diff
@@ -0,0 +1,16 @@
+class WorkflowEventFilterValidator < ActiveModel::Validator
+  SUPPORTED_FILTER_EVENTS = ['push', 'pull_request'].freeze
+
+  def validate(workflow)
+    # FIXME: replace private method calls
+    if workflow.send(:supported_filters).key?(:event)
+      # The filters method is needed for the branch_matches_branches_filter that will eventually be a validator.
+      workflow.errors.add(:base, "Workflow filter not supported: #{workflow.filters[:event]}") unless
```

Comment:
Error messages could be clearer if we split this up per event. So for example, if a webhook is for a push event, but the workflow should only run for pull requests, the error message could be quite specific like:

`Workflow $WORKFLOW_NAME ignored since the event filter 'push' doesn't match the webhook event.`